pkpdapp/pkpdapp/dash_apps/nca.py

```diff
@@ -0,0 +1,290 @@
+#
+
+import numpy as np
+import scipy.stats as stats
+
+
+class NCA():
+    def __init__(self,
+                 time, conc, DM,
+                 doseSchedule='Single', administrationRoute='IVBolus'
+                 ):
+        """
+        Initialise NCA class for calculating and storing NCA parameters
+        :param time: {np.ndarray} --- array of observation times
+        :param conc: {np.ndarray} --- array of observed concentration values
+        :param DM: {float} --- administered dose amount
+        :param doseSchedule: {string} Single or Multiple dose data
+        :param administrationRoute: {string} IVBolus, IVInfusion, ExtraVascular
+        """
+        # Check if intial concentration has been measured
+
+        if time[0] == 0:
+            self.is_c0extrapolated = False  # initial concentration measured
+            self.time = time
+            self.conc = conc
+            self.DM = DM  # dose amount
+            self.C_0 = None  # not extrapolated
+            self.administrationRoute = administrationRoute
+        else:
+            self.is_c0extrapolated = True  # initial concentration NOT measured
+            c0 = self.extrapolate_c0(conc, time)  # extrapolate initial conc
+            self.C_0 = c0
+            self.time = np.insert(time, 0, 0, axis=0)
+            self.conc = np.insert(conc, 0, c0, axis=0)
+            self.DM = DM  # dose amount
+            self.administrationRoute = administrationRoute
+
+    def calculate_nca(self):
+        """
+        Calculates and stores all single dose NCA parameters
+        :return: None
+        """
+        self.AUC_0_last = self._AUC_0_last()
+        self.AUMC_0_last = self._AUMC_0_last()
+        self.Lambda_z, self.R2, self.Num_points = self._Lambda_z()
+
+        self.AUC_infinity = self._AUC_infinity()
+        self.AUC_infinity_dose = self._AUC_infinity_dose()
+        # self.AUCx_y
+        self.AUC_extrap_percent = self._AUC_extrap_percent()
+        self.CL = self._CL()
+        self.C_max, self.T_max = self._find_Cmax()
+        self.C_max_Dose = self._C_max_Dose()
+        # C_max_x_y
+
+        self.AUMC = self._AUMC()
+        self.AUMC_extrap_percent = self._AUMC_extrap_percent()
+        self.MRT = self._MRT()
+        self.Tlast = self._Tlast()  # ?? above LOQ?
+        self.T_half = np.log(2) / self.Lambda_z
+        # T_max_x_y
+        if self.administrationRoute == 'IVBolus':
+            self.V_ss = self._V_ss()
+        else:
+            self.V_ss = None
+
+        self.V_z = self._V_z()
+
+    @staticmethod
+    def extrapolate_c0(y, x):
+        """
+        Calculate initial concentration when not given, computed using a (log)
+        regression of the first two data points in a profile.
+        Arguments:
+        y {np.ndarray} -- y coordinates of points on curve
+        x {np.ndarray} -- x coordinates of points on curve
+
+        Returns:
+        c0 {double} -- extrapoloated c0
+        """
+        slope, intercept, r_value, _, _ = \
```

Comment:
slope and r_value don't seem to be being used anywhere? So maybe just _ for them too?